app/lib/account/consent_backend.dart

```diff
@@ -219,9 +220,16 @@ class ConsentBackend {
 
   /// Returns the [Consent] for [consentId] and checks if it is for [user].
   Future<Consent> _lookupAndCheck(String consentId, User user) async {
-    final c = await _db.lookupValue<Consent>(
-        _db.emptyKey.append(Consent, id: consentId),
-        orElse: () => null);
+    final key = _db.emptyKey.append(Consent, id: consentId);
+    final c = await withRetryTransaction(_db, (tx) async {
```

Comment:
Just make this a `return await withRetryTransaction`, since we can throw `ResponseException` from within it.